packages/tx-construction/test/tx-builder/TxBuilder.test.ts

```diff
@@ -356,6 +393,23 @@ describe('GenericTxBuilder', () => {
       it('legit output with valid with address and coin', async () => {
         await expect(txBuilder.buildOutput().address(address).coin(output1Coin).build()).resolves.toBeTruthy();
       });
+
+      it('resolves handle to address', async () => {
+        const outputBuilderTx = await txBuilder
+          .buildOutput()
+          .address(address)
+          .handle('alice')
+          .coin(output1Coin)
+          .build();
+
+        expect(outputBuilderTx.handle).toBe(resolvedHandle);
+      });
+
+      it('rejects with an error when handle resolution fails', async () => {
```

Comment:
Missing a test for when handle provider resolves to `null` - no error, but also not resolved. Should probably create a new error type for that.
```suggestion
      it('rejects with an error when handle is not found')
      it('rejects with an error when handle resolution fails', async () => {
```

Reply:
I think the type was wrong and the provider will either return an array of `HandleResolution` or throw an error. At least that is how it is in its current implementation.

Reply:
No, the type intentionally returns an array of either `null | HandleResolution`, where `null` means that it couldn't be resolved. This enables the user to interpret the result better, e.g.
```typescript
resolveHandles(['handle1', 'handle2']) // => [null, {...}]
```